libraries/ts/src/margin/marginAccount.ts

```diff
@@ -659,7 +659,11 @@ export class MarginAccount {
       seed = await this.getUnusedAccountSeed({ programs, provider, owner })
     }
     const marginAccount = new MarginAccount(programs, provider, owner, seed, pools, walletTokens)
-    await marginAccount.createAccount()
+    const instructions: TransactionInstruction[] = []
+    await marginAccount.withCreateAccount(instructions)
+    if (instructions.length > 0) {
+      return await provider.sendAndConfirm(new Transaction().add(...instructions))
+    }
```

Comment:
This does the exact same thing as `marginAccount.createAccount()` so you can leave it as is
```suggestion
    await marginAccount.createAccount()
```